### ISSUE TITLE
bug: Unable to setup server with AddApolloFederationV2 

### ISSUE THREAD

**Issue by StefanGies99** [apollographql/federation-hotchocolate#35]
**Description of the bug**

I'm trying to add federation v2 to my Hotchocolate application and I'm running into the following issue and not sure why it's causing issues. 

```
Unhandled exception. System.InvalidOperationException: No compatible constructor found for input type type `ApolloGraphQL.HotChocolate.Federation.Two.Link`.
Either you have to provide a public constructor with settable properties or a public constructor that allows to pass in values for read-only properties. There was no way to set the following properties: .
   at HotChocolate.Utilities.Serialization.InputObjectConstructorResolver.GetCompatibleConstructor[T](Type type, FieldCollection`1 fields, Dictionary`2 fieldMap, HashSet`1 required)
   at HotChocolate.Utilities.Serialization.InputObjectCompiler.CompileFactory(DirectiveType directiveType, ConstructorInfo constructor)
   at HotChocolate.Types.DirectiveType.OnCompleteCreateInstance(ITypeCompletionContext context, DirectiveTypeDefinition definition)
   at HotChocolate.Types.DirectiveType.OnCompleteType(ITypeCompletionContext context, DirectiveTypeDefinition definition)
   at HotChocolate.Types.TypeSystemObjectBase`1.CompleteType(ITypeCompletionContext context)
   at HotChocolate.Configuration.TypeInitializer.CompleteType(RegisteredType registeredType)
   at HotChocolate.Configuration.TypeInitializer.<CompleteTypes>b__29_0(RegisteredType type)
   at HotChocolate.Configuration.TypeInitializer.ProcessTypes(TypeDependencyFulfilled fulfilled, Func`2 action)
   at HotChocolate.Configuration.TypeInitializer.CompleteTypes()
   at HotChocolate.Configuration.TypeInitializer.Initialize()
   at HotChocolate.SchemaBuilder.Setup.InitializeTypes(SchemaBuilder builder, IDescriptorContext context, IReadOnlyList`1 types)
   at HotChocolate.SchemaBuilder.Setup.Create(SchemaBuilder builder, LazySchema lazySchema, IDescriptorContext context)
   at HotChocolate.SchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.SchemaBuilder.HotChocolate.ISchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaAsync(ConfigurationContext context, RequestExecutorSetup setup, RequestExecutorOptions executorOptions, IServiceProvider schemaServices, TypeModuleChangeMonitor typeModuleChangeMonitor, CancellationToken cancellationToken)
```

I've tried all possible versions that are available for v2 of federation but none of them work. The only federation version that I get working is v1 of federation.

**Comment by dariuszkuc** [comment]
Hello 👋 
Can you provide a link to a Github repository that reproduces this issue?

**Comment by StefanGies99** [comment]
Let me figure out something! I'll get back to you later today!

**Comment by StefanGies99** [comment]
I managed to figure out what is causing the issue, as soon as you set the `DefaultBindingBehavior` to be explicit it fails to compile with the error mentioned above.

```
builder.Services
    .AddGraphQLServer()
    .AddApolloFederationV2(new CustomSchema())
    .AddType<CustomDirectiveType>()
    .AddType<Inventory>()
    .AddQueryType<Query>()
    .ModifyOptions(options => options.DefaultBindingBehavior = BindingBehavior.Explicit) <- I added this line of code to the Compatibility project and that causes the issue
    .RegisterService<Data>();
```

**Comment by StefanGies99** [comment]
Hi @dariuszkuc I really appreciate the effort that you've taken to fix this bug!! 
Could you give an estimate when a new version will be released so that I can check my codebase if everything works?

**Comment by dariuszkuc** [comment]
released v[1.2.0](https://github.com/apollographql/federation-hotchocolate/releases/tag/v1.2.0) with the fix
